src/ui/public/visualize/loader/visualize_data_loader.ts

```diff
@@ -71,74 +67,43 @@ export class VisualizeDataLoader {
-        filters: params.filters
-          ? params.filters.filter(filter => !filter.meta.disabled)
-          : undefined,
-      });
-
-      // No need to call the response handler when there have been no data nor has been there changes
-      // in the vis-state (response handler does not depend on uiStat
-      const canSkipResponseHandler =
-        this.previousRequestHandlerResponse &&
-        this.previousRequestHandlerResponse === requestHandlerResponse &&
-        this.previousVisState &&
-        isEqual(this.previousVisState, this.vis.getState());
-
-      this.previousVisState = this.vis.getState();
-      this.previousRequestHandlerResponse = requestHandlerResponse;
-
-      if (!canSkipResponseHandler) {
-        this.visData = await Promise.resolve(
-          this.responseHandler(requestHandlerResponse, this.vis.params.dimensions)
-        );
-      }
-
-      return {
-        as: 'visualization',
-        value: {
-          visType: this.vis.type.name,
-          visData: this.visData,
-          visConfig: this.vis.params,
-          params: {},
-        },
-      };
-    } catch (error) {
-      params.searchSource.cancelQueued();
-
-      this.vis.requestError = error;
-      this.vis.showRequestError =
-        error.type && ['NO_OP_SEARCH_STRATEGY', 'UNSUPPORTED_QUERY'].includes(error.type);
-
-      // tslint:disable-next-line
-      console.error(error);
-
-      if (isTermSizeZeroError(error)) {
-        toastNotifications.addDanger(
-          `Your visualization ('${this.vis.title}') has an error: it has a term ` +
-            `aggregation with a size of 0. Please set it to a number greater than 0 to resolve ` +
-            `the error.`
-        );
-        return;
-      }
-
-      toastNotifications.addDanger({
-        title: 'Error in visualization',
-        text: error.message,
-      });
+    // searchSource is only there for courier request handler
+    const requestHandlerResponse = await this.requestHandler({
+      partialRows: this.vis.params.partialRows || this.vis.type.requiresPartialRows,
+      isHierarchical: this.vis.isHierarchical(),
+      visParams: this.vis.params,
+      ...params,
+      filters: params.filters ? params.filters.filter(filter => !filter.meta.disabled) : undefined,
+    });
+
+    // No need to call the response handler when there have been no data nor has been there changes
+    // in the vis-state (response handler does not depend on uiStat
+    const canSkipResponseHandler =
+      this.previousRequestHandlerResponse &&
+      this.previousRequestHandlerResponse === requestHandlerResponse &&
+      this.previousVisState &&
+      isEqual(this.previousVisState, this.vis.getState());
+
+    this.previousVisState = this.vis.getState();
+    this.previousRequestHandlerResponse = requestHandlerResponse;
+
+    if (!canSkipResponseHandler) {
+      this.visData = await Promise.resolve(
+        this.responseHandler(requestHandlerResponse, this.vis.params.dimensions)
+      );
```

Comment:
Since errors are now caught in embedded visualize handler, the `try/catch` was no longer necessary here.